locator/token_metadata.cc

```diff
@@ -1261,13 +1259,33 @@ topology::topology(config cfg)
     _pending_locations[utils::fb_utilities::get_broadcast_address()] = std::move(cfg.local_dc_rack);
 }
 
-topology::topology(const topology& other)
-        : _dc_endpoints(other._dc_endpoints)
-        , _dc_racks(other._dc_racks)
-        , _current_locations(other._current_locations)
-        , _pending_locations(other._pending_locations)
-        , _sort_by_proximity(other._sort_by_proximity)
-{
+future<topology> topology::clone_gently() const {
```

Comment:
suggestion for incremental change -- worth extending the util/stall_free.hh with cloning?

Reply:
care to provide a more specific example? I don't follow exactly where you're aiming to...

Reply:
There's a bunch of polymorphic helpers in this header that can gently clean pretty much anything iterable. My suggestion is that it might makes sense to have gentle cloning helpers in this file.

Reply:
interesting idea, sounds feasible